app/main.py

```diff
@@ -1,2 +1,91 @@
+from random import randint
+from typing import Any
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    node_key: int | float | str | tuple
+    node_hash: int
+    node_value: Any
```

Comment:
Why do you need to specify node_ here in every attribute? I think, its not necessary, because you anyway will take this attibute from a Node type. Also you don't have the same in dictionary like dictionary_length. What do you think?